packages/loader/web-code-loader/src/webLoader.ts

```diff
@@ -60,10 +61,13 @@ export class WebCodeLoader implements ICodeLoader {
         }
 
         const fluidModuleP = this.loadModuleFromResolvedCodeDetails(resolved);
+
         if (resolved.resolvedPackageCacheId !== undefined) {
             this.loadedModules.set(resolved.resolvedPackageCacheId, fluidModuleP);
         }
-        return fluidModuleP;
+        const fluidModule = await fluidModuleP;
+
+        return fluidModule;
```

Comment:
Returning the promise should be fine here, since we don't need to modify the module before return.

```suggestion
        return fluidModuleP;
```